packages/block-editor/src/components/link-control/test/index.js

```diff
@@ -450,17 +448,52 @@ describe( 'Default search suggestions', () => {
 
 		await eventLoopTick();
 
-		searchResultElements = container.querySelectorAll(
+		const searchResultElements = container.querySelectorAll(
 			'[role="listbox"] [role="option"]'
 		);
 
-		expect( searchResultElements ).toHaveLength( 0 );
+		const searchInput = container.querySelector(
+			'input[aria-label="URL"]'
+		);
 
-		expect( mockFetchSearchSuggestions ).not.toHaveBeenCalled();
+		// search input is set to the URL value
+		expect( searchInput.value ).toEqual( fauxEntitySuggestions[ 0 ].url );
 
-		//
-		// Reset the search to empty and check the initial suggestions are now shown.
-		//
+		// it should match any url that's like ?p= and also include a URL option
+		expect( searchResultElements ).toHaveLength( 5 );
+
+		expect( mockFetchSearchSuggestions ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	it( 'should display initial suggestions when input value is reset', async () => {
+		const searchTerm = 'Hello world';
+
+		act( () => {
+			render( <LinkControl showInitialSuggestions />, container );
+		} );
+
+		let searchResultElements;
+		let searchInput;
+
+		// Search Input UI
+		searchInput = container.querySelector( 'input[aria-label="URL"]' );
+
+		// Simulate searching for a term
+		act( () => {
+			Simulate.change( searchInput, { target: { value: searchTerm } } );
+		} );
+
+		// fetchFauxEntitySuggestions resolves on next "tick" of event loop
+		await eventLoopTick();
+
+		expect( searchInput.value ).toBe( searchTerm );
+
+		// TODO: select these by aria relationship to autocomplete rather than arbitary selector.
+		searchResultElements = container.querySelectorAll(
+			'[role="listbox"] [role="option"]'
+		);
+
+		// Grab the reset button now it's available
```

Comment:
As well as a testing for clicking on the reset UI I think we should test what happens when the user simply **deletes** the value in the input manually. Just trigger a change on the `input` with a empty string value.

Reply:
Added a test for this. 👍

Reply:
Nice 👍